src/mono/System.Private.CoreLib/src/System/Threading/PreAllocatedOverlapped.cs

```diff
@@ -8,5 +8,6 @@ public sealed class PreAllocatedOverlapped : System.IDisposable
         [System.CLSCompliantAttribute(false)]
         public PreAllocatedOverlapped(System.Threading.IOCompletionCallback callback, object? state, object? pinData) { }
         public void Dispose() { }
+        internal bool IsUserObject(byte[]? buffer) => false;
```

Comment:
@marek-safar would you be the best person to take a look at any changes we make in the `System.IO` space in mono?

Reply:
Does it need to be Mono specific?

@lambdageek @CoffeeFlux to approve

Reply:
Yeah, there's no special runtime handling from what I can see; the class is just a stub on Mono. I'm confused how this implementation is passing tests on Mono if it relies on that working. Maybe we just had them disabled anyway?

I haven't gone through the PR in detail since it's so large, but if the rewrite is desirable long-term and support for `PreAllocatedOverlapped` is necessary, we can potentially add that. It seems to be Windows-only though, so I don't think it's urgent. I'll open an issue.